src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -340,44 +341,57 @@ Transactor::checkSign (PreclaimContext const& ctx)
 NotTEC
 Transactor::checkSingleSign (PreclaimContext const& ctx)
 {
-    auto const id = ctx.tx.getAccountID(sfAccount);
-
-    auto const sle = ctx.view.read(
-        keylet::account(id));
-    auto const hasAuthKey     = sle->isFieldPresent (sfRegularKey);
-
-    // Consistency: Check signature & verify the transaction's signing
-    // public key is authorized for signing.
-    auto const spk = ctx.tx.getSigningPubKey();
-    if (!publicKeyType (makeSlice (spk)))
+    // Check that the value in the signing key slot is a public key.
+    auto const pkSigner = ctx.tx.getSigningPubKey();
+    if (!publicKeyType(makeSlice(pkSigner)))
     {
         JLOG(ctx.j.trace()) <<
             "checkSingleSign: signing public key type is unknown";
         return tefBAD_AUTH; // FIXME: should be better error!
     }
 
-    auto const pkAccount = calcAccountID (
-        PublicKey (makeSlice (spk)));
+    // Look up the account.
+    auto const idSigner = calcAccountID(PublicKey(makeSlice(pkSigner)));
+    auto const idAccount = ctx.tx.getAccountID(sfAccount);
+    auto const sleAccount = ctx.view.read(keylet::account(idAccount));
+    auto const isMasterDisabled = sleAccount->isFlag(lsfDisableMaster);
+
+    if (ctx.view.rules().enabled(fixDisabledRegularKey)) {
+
```

Comment:
nit: spacing/indentation around here looks a little weird. (I think `{` should be on a newline).

Reply:
Oh, yeah, I missed that. Our style guide requires `{` to always be on a newline except for namespaces.